### PR TITLE
import and export cluster templates

### LINKED ISSUE #7435: Import/export templates from file
**User Story**
User should be able to import or export templates from the json/yaml file.

### PR DESCRIPTION
**Pull request by zreigz**
**What does this PR do / Why do we need it**: Two methods to export and import cluster templates to/from file

**Does this PR close any issues?**:
Fixes #7435

```release-note
Add possibilities to import and export cluster templates from/to file
```

### CODE REVIEW

File: pkg/handler/v2/routes_v2.go [hunk 1]
```diff
@@ -4287,13 +4293,41 @@ func (r Routing) createClusterTemplate() http.Handler {
...
+		)(clustertemplate.CreateEndpoint(r.projectProvider, r.privilegedProjectProvider, r.userInfoGetter, r.clusterTemplateProvider, r.seedsGetter, r.presetProvider, r.caBundle, r.exposeStrategy, r.sshKeyProvider, r.kubermaticConfigGetter, r.features)),
 		clustertemplate.DecodeCreateReq,
 		handler.SetStatusCreatedHeader(handler.EncodeJSON),
 		r.defaultServerOptions()...,
 	)
 }
 
+// swagger:route POST /api/v2/projects/{project_id}/clustertemplates/import project importClusterTemplate
+//
+//     Import a cluster templates for the given project.
```
Comment (lsviben) [inline pkg/handler/v2/routes_v2.go:4305]:
```suggestion
//     Import a cluster template for the given project.
```

File: pkg/handler/v2/cluster_template/cluster_template.go [hunk 2]
```diff
@@ -397,6 +487,43 @@ func CreateInstanceEndpoint(projectProvider provider.ProjectProvider, privileged
...
+}
+
+func EncodeClusterTemplate(_ context.Context, w http.ResponseWriter, response interface{}) (err error) {
+	rsp := response.(*encodeClusterTemplateResponse)
+	clusterTemplate := rsp.clusterTemplate
+	filename := "clusterTemplate"
+
+	if len(rsp.filePrefix) > 0 {
+		filename = fmt.Sprintf("%s-%s", filename, rsp.filePrefix)
+	}
```
Comment (lsviben) [inline pkg/handler/v2/cluster_template/cluster_template.go]: if its prefix, it should go infront

File: pkg/handler/v2/cluster_template/cluster_template.go [hunk 3]
```diff
@@ -456,6 +583,34 @@ func DecodeGetReq(c context.Context, r *http.Request) (interface{}, error) {
...
+
+	getReq, err := DecodeGetReq(c, r)
+	if err != nil {
+		return nil, err
+	}
+	req.getClusterTemplatesReq = getReq.(getClusterTemplatesReq)
+
+	queryParam := r.URL.Query().Get("format")
+
+	if len(queryParam) > 0 && queryParam != yamlFormat {
```
Comment (lsviben) [inline pkg/handler/v2/cluster_template/cluster_template.go]: what if someone sets format `json`. It should work but here it will get denied

Comment (zreigz) [reply pkg/handler/v2/cluster_template/cluster_template.go]: yes, you are right, thanks

File: pkg/handler/v2/cluster_template/cluster_template.go (hunk 2 above)
Comment (zreigz) [reply pkg/handler/v2/cluster_template/cluster_template.go]: we also have such patter for other resources and also for kubeconfig so I decided  to don't change it

Comment (lsviben) [reply pkg/handler/v2/cluster_template/cluster_template.go]: ok i see. Rename it to fileSuffix then

**Comment by zreigz** [comment]
@lsviben PTAL

**Pull request merged**